frontend/packages/ceph-storage-plugin/src/components/modals/volume-snapshot-modal/volume-snapshot-modal.tsx

```diff
@@ -37,39 +46,99 @@ export type VolumeSnapshotModalProps = {
 
 export const VolumeSnapshotModal = withHandlePromise((props: VolumeSnapshotModalProps) => {
   const { close, cancel, pvcData, errorMessage, inProgress, handlePromise } = props;
+  const current = new Date();
   const resource = pvcData.data as K8sResourceKind;
   const [snapshotCount, setSnapshotCount] = React.useState(3); // For all schedules
-  const [snapshotWeek, setSnapshotWeek] = React.useState(0); // For Weekly schedule
-  const [snapshotTime, setSnapshotTime] = React.useState('00:01'); // For weekly and monthly
-  const [snapshotMonth, setSnapshotMonth] = React.useState(1); // For monthly schedule
-  const [snapshotName, setSnapshotName] = React.useState<string>();
+
+  const [snapshotName, setSnapshotName] = React.useState<string>(
+    `${getName(resource) || 'pvc'}-snapshot`,
+  );
   const [scheduleLabel, setScheduleLabel] = React.useState<string>();
   const [scheduleType, setScheduleType] = React.useState(snapshotTypes.Single);
+  const [scheduleTimezone, setScheduleTimezone] = React.useState<string>(
+    `${momentTZ.tz.names().indexOf(momentTZ.tz.guess())}`,
+  );
+  const [snapshotDayOfMonth, setSnapshotDayOfMonth] = React.useState(
+    momentTZ.tz(current, momentTZ.tz.names()[scheduleTimezone]).date() - 1,
+  ); // For monthly schedule
+  const [snapshotTime, setSnapshotTime] = React.useState(
+    `${
+      momentTZ.tz(current, momentTZ.tz.names()[scheduleTimezone]).hour() < 10
+        ? `0${momentTZ.tz(current, momentTZ.tz.names()[scheduleTimezone]).hour()}`
+        : momentTZ.tz(current, momentTZ.tz.names()[scheduleTimezone]).hour()
+    }:${
+      momentTZ.tz(current, momentTZ.tz.names()[scheduleTimezone]).minute() < 10
+        ? `0${momentTZ.tz(current, momentTZ.tz.names()[scheduleTimezone]).minute()}`
+        : momentTZ.tz(current, momentTZ.tz.names()[scheduleTimezone]).minute()
```

Comment:
You should store the value of `momentTZ.tz.names()[scheduleTimezone]` in a variable and use it across all places. It will look cleaner.